protocol/pubsub/v2/message_test.go

```diff
@@ -62,6 +63,22 @@ func TestFinish(t *testing.T) {
 			err:     fmt.Errorf("error"),
 			wantErr: true,
 		},
+		{
+			name: "result not acked",
+			pm: &pubsub.Message{
+				ID: "testid",
+			},
+			err:     protocol.NewReceipt(false, "error"),
+			wantErr: true,
+		},
+		{
+			name: "result acked",
+			pm: &pubsub.Message{
+				ID: "testid",
+			},
+			err:     protocol.NewReceipt(true, "error"),
+			wantErr: false,
```

Comment:
see other discussion: IMHO we want to surface the error here or is it already handled/logged somewhere before?

Reply:
The point is, I think that, when returning protocol.Result instead of error, that's much more a "state" than a real error. But, we could keep the error and just mitigate pubsub ack behaviour.
(In my case, I would for sure disable cloudevents logging in that case but it's fine to me)

Reply:
`protocol.Result` is of type `error` and implements `error` specific [logic](https://github.com/cloudevents/sdk-go/blob/972dfc7354b54f0a7dc6929a1a47930062a8596e/v2/protocol/result.go) e.g., assertions so I would classify it as `error` because that's how users of the SDK can use it (uses error wrapping internally). Please see other comment regarding returning `error` and adding a test.